internal/hooks/tracking.go

```diff
@@ -135,8 +135,17 @@ func MarkAsOkToDelete(ctx context.Context, c client.Client, obj client.Object) e
 }
 
 func addToCommaSeparatedList(list string, items ...string) string {
-	set := sets.NewString(strings.Split(list, ",")...)
+	set := sets.NewString()
+	if list != "" {
+		set.Insert(strings.Split(list, ",")...)
+	}
```

Comment:
Is the `if` necessary given we're deleting the empty strings in any case?

Reply:
Hm no not really. I think I added it one after another and thought we'll get rid of the set.Delete again at some point.

But let's simplify